boards/PDM/Src/Io/Io_Efuse_22XS4200.c

```diff
@@ -0,0 +1,198 @@
+static SPI_HandleTypeDef *efuse_spi_handle;
+
+void Io_Efuse_22XS4200_Init(SPI_HandleTypeDef *const hspi)
+{
+    efuse_spi_handle = hspi;
+}
+
+void Io_Efuse_22XS4200_configureEfuse(struct E_Fuse *e_fuse)
+{
+    // Global config register
+    Io_Efuse_22XS4200_WriteReg(SI_GCR_ADDR, GCR_CONFIG, e_fuse);
+
+    // Channel 0 config registers
+    Io_Efuse_22XS4200_WriteReg(
+            SI_RETRY_0_ADDR, RETRY_0_CONFIG, e_fuse);
+    Io_Efuse_22XS4200_WriteReg(
+            SI_CONFR_0_ADDR, CONFR_0_CONFIG, e_fuse);
+    Io_Efuse_22XS4200_WriteReg(SI_OCR_0_ADDR, OCR_0_CONFIG, e_fuse);
+
+    // Channel 1 config registers
+    Io_Efuse_22XS4200_WriteReg(
+            SI_RETRY_1_ADDR, RETRY_1_CONFIG, e_fuse);
+    Io_Efuse_22XS4200_WriteReg(
+            SI_CONFR_1_ADDR, CONFR_1_CONFIG, e_fuse);
+    Io_Efuse_22XS4200_WriteReg(SI_OCR_1_ADDR, OCR_1_CONFIG, e_fuse);
+}
+
+void Io_Efuse_22XS4200_configure_channel_monitoring(
+        uint8_t        selection,
+        struct E_Fuse *e_fuse)
+{
+    uint16_t reg_val = 0x0000;
+
+    // Read original content of GCR Register
+    reg_val = Io_Efuse_22XS4200_ReadReg(SI_GCR_ADDR, e_fuse);
+
+    reg_val &= ~(CSNS1_EN_MASK | CSNS0_EN_MASK);
+    reg_val |= (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK));
+
+    Io_Efuse_22XS4200_WriteReg(SI_GCR_ADDR, reg_val, e_fuse);
+}
+
+void Io_Efuse_22XS4200_UpdateStatus(struct E_Fuse *e_fuse)
+{
+    uint16_t status = StatusType_NoFault;
+
+    status = Io_Efuse_22XS4200_ReadReg(SO_STATR_ADDR, e_fuse);
+
+    /* Clear other bits */
+    status &= EFUSE_SO_DATA_MASK;
+
+    e_fuse->status = (StatusType_e)status;
+}
+
+void Io_Efuse_22XS4200_UpdateFaults(struct E_Fuse *e_fuse)
+{
+    uint16_t channel_0_faults = FaultType_NoFault;
+    uint16_t channel_1_faults = FaultType_NoFault;
+
+    channel_0_faults = Io_Efuse_22XS4200_ReadReg(SO_FAULTR_0_ADDR, e_fuse);
+    channel_1_faults = Io_Efuse_22XS4200_ReadReg(SO_FAULTR_1_ADDR, e_fuse);
+
+    /* Clear other bits */
+    channel_0_faults &= EFUSE_SO_DATA_MASK;
+    channel_1_faults &= EFUSE_SO_DATA_MASK;
+
+    e_fuse->faults.channel_0_faults = (FaultType_e)channel_0_faults;
+    e_fuse->faults.channel_1_faults = (FaultType_e)channel_1_faults;
+}
+
+void Io_Efuse_22XS4200_WriteReg(
+        uint8_t        register_address,
+        uint16_t       register_value,
+        struct E_Fuse *e_fuse)
+{
+    HAL_StatusTypeDef status;
+    uint16_t          command    = 0x0000U;
+    uint8_t           parity_bit = 0;
+
+    command =
+            (uint16_t)((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);
+    /* place register value to be written (bits 0->8) */
+    command = (uint16_t)(command | (register_value & EFUSE_SI_DATA_MASK));
+    /* invert watchdog bit state */
+    if (e_fuse->watch_dog_state)
+    {
+        command |= WATCH_DOG_BIT;
+    }
+    else
+    {
+        command &= ~(WATCH_DOG_BIT);
+    }
+    /* invert watchdog bit state for next write */
+    e_fuse->watch_dog_state = !e_fuse->watch_dog_state;
```

Comment:
The parity bit must be set if the parity is un-even:
![image](https://user-images.githubusercontent.com/25499626/81438503-bc75b380-9121-11ea-88d2-b4838944e4c3.png)